### PR TITLE
Feature: add info price pin 

### LINKED ISSUE #279: Add Information Pin in Activity Details
- [x] Place an information pin in the activity details screen to inform users that payment for the activity will occur upon attendance. 

- [x] Ensure the information is clearly visible and contextually relevant.

### PR DESCRIPTION
**Pull request by hmaryam84**
### **Pull Request: Enhance Payment Information Distinctions**

This PR introduces a new feature and tests to improve the user experience related to payments . The changes are outlined below, grouped by functionality for clarity:

---

#### 1. Add Payment Information Pin
- **Dialog Added**: Introduced an information dialog explaining how the payment process will work.
  - The dialog provides clarity to users about payment expectations during activities.
  - Improves transparency for both free and non-free activities.

---

#### 2. Distinction Between Free and Non-Free Activities
- **Dynamic Text**: Updated the information dialog to display content based on the price of the activity.
  - **For Free Activities**:
    - The dialog clearly indicates that no payment is required.
  - **For Paid Activities**:
    - Users are reminded to bring the specified amount for the activity.

---

### **Tests**
- Created tests specifically for:
  - Verifying the display and behavior of the payment information dialog.
  - Ensuring the button triggering the dialog works as intended.

Feel free to tweak the formatting or details further. Let me know if you need additional assistance!

close #279

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -18,6 +18,10 @@
     <string name="no_internet_connection">No internet connection</string>
     <string name="internet_connection_ask">Please check your connection and try again</string>
     <string name="button_to_map">See on map</string>
+    <string name="ok">OK</string>
```
Comment (mohamedtahaguelzim) [inline app/src/main/res/values/strings.xml:21]: this ok message should be more explanatory for its name

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -18,6 +18,10 @@
...
+    <string name="payment_info">Payment Information</string>
+    <string name="payment_explanation">The payment will be done on-site. Please ensure you have the necessary amount ready.</string>
+    <string name="free_activity">This activity is free. No payment is needed to attend.</string>
```
Comment (mohamedtahaguelzim) [inline app/src/main/res/values/strings.xml:24]: great effort! very useful for future maintenance and revision

File: app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt
Comment (mohamedtahaguelzim) [inline app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt:1]: You added many testTags, which ensure a good testing capability. Great!

File: app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt [hunk 3]
```diff
@@ -235,12 +239,14 @@ fun ActivityDetailsScreen(
...
                   verticalAlignment = Alignment.CenterVertically,
-                  modifier = Modifier.testTag("price")) {
+                  modifier = Modifier.testTag("price").fillMaxWidth()) {
                     Icon(Icons.Filled.AttachMoney, contentDescription = "Price")
                     Spacer(modifier = Modifier.width(SMALL_PADDING.dp))
                     Text(
                         text = if (price != null) "${price.toString()} CHF" else "not defined yet",
                         modifier = Modifier.testTag("priceText"))
+                    Spacer(modifier = Modifier.weight(WIDTH_FRACTION))
+                    PaymentInfoScreen(price ?: 0.0)
```
Comment (mohamedtahaguelzim) [inline app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt:249]: Super handling of the null price case. Null price would imply a free activity. This works as we expect.

File: app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt [hunk 4]
```diff
@@ -672,6 +678,60 @@ fun CommentItem(
...
+  var showDialog by remember { mutableStateOf(false) }
+
+  // Payment Section
+  Row(
+      verticalAlignment = Alignment.CenterVertically,
+      horizontalArrangement = Arrangement.Start,
+      modifier = Modifier.testTag("paymentSection")) {
+        // Payment Text
+        Text(
+            text = "Payment info",
```
Comment (mohamedtahaguelzim) [inline app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt:692]: Could have put this text in the res file too!

File: app/src/androidTest/java/com/android/sample/ui/activityDetails/ActivityDetailsScreenTest1.kt [hunk 5]
```diff
@@ -569,25 +570,57 @@ class ActivityDetailsScreenAndroidTest {
...
+    // Verify that the like button is toggled
+    composeTestRule.onNodeWithTag("likeButtontrue").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("likeButtontrue").performClick()
 
-      // Verify that the like button is toggled
-      composeTestRule.onNodeWithTag("likeButtonfalse").assertIsDisplayed()
-    }
+    // Verify that the like button is toggled
+    composeTestRule.onNodeWithTag("likeButtonfalse").assertIsDisplayed()
+  }
```
Comment (mohamedtahaguelzim) [inline app/src/androidTest/java/com/android/sample/ui/activityDetails/ActivityDetailsScreenTest1.kt:594]: This test looks the same to me. Wonder why git considers it as changed.

File: app/src/androidTest/java/com/android/sample/ui/activityDetails/ActivityDetailsScreenTest1.kt
Comment (mohamedtahaguelzim) [inline app/src/androidTest/java/com/android/sample/ui/activityDetails/ActivityDetailsScreenTest1.kt:1]: Great component tests! Did you have issues testing it screen-wide ? If you believe that testing it screen-wide is useless, in terms of code coverage, then ok. Else, you should consider it. Nonetheless, we will most definitely test this in the e2e tests, which can replace the screen tests.

File: app/src/androidTest/java/com/android/sample/ui/activityDetails/ActivityDetailsScreenTest1.kt [hunk 6]
```diff
@@ -569,25 +570,57 @@ class ActivityDetailsScreenAndroidTest {
...
+  @Test
+  fun paymentInfoDialog() {
+    composeTestRule.setContent { PaymentInfoScreen(10.0) }
+    composeTestRule.onNodeWithTag("infoIconButton").performClick()
+    composeTestRule.onNodeWithTag("paymentInfoDialog").assertExists()
+    composeTestRule.onNodeWithTag("paymentInfoTitle").assertExists()
+    composeTestRule.onNodeWithTag("paymentInfoText").assertExists()
+    composeTestRule.onNodeWithTag("okButton").performClick()
+    composeTestRule.onNodeWithTag("infoDialog").assertDoesNotExist()
+  }
```
Comment (mohamedtahaguelzim) [inline app/src/androidTest/java/com/android/sample/ui/activityDetails/ActivityDetailsScreenTest1.kt:614]: Looks good to me !

Comment (mohamedtahaguelzim) [reply app/src/androidTest/java/com/android/sample/ui/activityDetails/ActivityDetailsScreenTest1.kt:614]: Could have also checked that the freeInfoText does not exist/appear.

File: app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt [hunk 7]
```diff
@@ -672,6 +678,60 @@ fun CommentItem(
...
+              contentDescription = "Info",
+              tint = Color.Gray)
+        }
+      }
+
+  // Info Dialog
+  if (showDialog) {
+    AlertDialog(
+        modifier = Modifier.testTag("paymentInfoDialog"),
+        onDismissRequest = { showDialog = false },
```
Comment (mohamedtahaguelzim) [inline app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt:709]: Great that you added this, so that we can click outside the dialog to close it.

File: app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt [hunk 8]
```diff
@@ -672,6 +678,60 @@ fun CommentItem(
...
+            Text(text = stringResource(id = R.string.ok))
+          }
+        },
+        title = {
+          Text(
+              modifier = Modifier.testTag("paymentInfoTitle"),
+              text = stringResource(id = R.string.payment_info))
+        },
+        text = {
+          if (price != 0.0) {
```
Comment (mohamedtahaguelzim) [inline app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt:721]: I guess you assume no price could be negative. It may would have been better to consider this edge case

Comment (mohamedtahaguelzim) [reply app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt:721]: You could for example show a message that says _"Invalid Price"_ when `price < 0.0`

**Review by mohamedtahaguelzim — COMMENTED** [review]
Great! Thanks for you highly appreciated work. This PR is fundamental in our new dev goals : refining the app and prioritising UX. Some small edge cases could have been considered, but they do have very little to no impact on our app. There are no screen-wide UI tests. We should keep in mind to cover this in futur e2e tests.

**Review by mohamedtahaguelzim — APPROVED** [review]
